circle.yml

```diff
@@ -83,10 +78,57 @@ jobs:
           name: Run Tests
           command: |
             . venv/bin/activate
-            bin/run-circleci-tests
+            mkdir -p /tmp/coverage/$CIRCLE_SHA1
+            make << parameters.tests_type >>
+            if [ -e .coverage.* ]; then
+                cp .coverage.* /tmp/coverage/$CIRCLE_SHA1/.
+            fi
+            if [ -d coverage ]; then
+                cp -R coverage/* /tmp/coverage/$CIRCLE_SHA1/.
+            fi
           no_output_timeout: 47m
           environment:
             ANSIBLE_HOST_KEY_CHECKING: False
+      - persist_to_workspace:
+          root: /tmp
+          paths:
+              - coverage
+  coverage:
+    docker:
+      - image: circleci/python:3.5.4-node
+    steps:
+      - checkout
+      - restore_cache:
+          key: dependencies-{{ checksum "requirements.txt" }}
+      - run:
+          name: Install Dependencies
+          command: |
+            python3 -m venv venv
+            . venv/bin/activate
+            pip install --upgrade pip
+            pip install --upgrade virtualenv
+            # The jessie-updates and jessie-backports areas are now removed from the repositories. So
+            # the lines corresponding to those areas have to be removed to fix the resulting errors.
+            # This can be removed after upgrading to the image based on Debian Stretch.
+            sudo sed -i '/jessie-updates/d' /etc/apt/sources.list
+            sudo sed -i '/jessie-backports/d' /etc/apt/sources.list
+            sudo apt-get update; sudo apt-get install python2.7-dev
+            sudo apt-get install mysql-client
+            pip install -r requirements.txt
+            pip install -r cleanup_utils/requirements.txt
+            sudo apt-get install postgresql-client
+            sudo apt-get install unzip
+            sudo wget -P /tmp https://releases.hashicorp.com/consul/1.2.1/consul_1.2.1_linux_amd64.zip
+            make install_system_dependencies
+            npm install
+      - attach_workspace:
+          at: /tmp/workspace
+      - run:
+          command: |
+              . venv/bin/activate
+              coverage combine /tmp/workspace/coverage/$CIRCLE_SHA1/
+              make cov.html
+              if [ -e /tmp/workspace/coverage/$CIRCLE_SHA1/text/coverage.txt ]; then cat /tmp/workspace/coverage/$CIRCLE_SHA1/text/coverage.txt; fi
```

Comment:
Just a question: the coverage for JavaScript is just being reported but isn't taken into account for the minimum, is that right?

Reply:
There's two minimums. One is calculated by `coverage.py` and another one is calculated by `istanbul`. The codecoverage.py minimum is set in the [coverage threshold](https://github.com/open-craft/opencraft/pull/426/files#diff-b67911656ef5d18c4ae36cb6741b7965R24) and the istanbul minimum is setup in the [karma.config file](https://github.com/open-craft/opencraft/pull/426/files#diff-a2a3b7b0c9c3b4b93b4aebf4e3ec3cfbR122).
There's no simple way to combine both of this (afaik). Services like codecov or coveralls do combine multiple reports in different formats.

Reply:
:+1: